tests/testthat/test-distribution-formulas.R

```diff
@@ -0,0 +1,89 @@
+    # when there are other options, this would be where the option would be chosen)
+    recruitment <- new(BevertonHoltRecruitment)
+
+    # set up log_rzero (equilibrium recruitment)
+    recruitment$log_rzero[1]$value <- log(om_input$R0)
+    recruitment$log_rzero[1]$is_random_effect <- FALSE
+    recruitment$log_rzero[1]$estimated <- TRUE
+    # set up logit_steep
+    recruitment$logit_steep[1]$value <- -log(1.0 - om_input$h) + log(om_input$h - 0.2)
+    recruitment$logit_steep[1]$is_random_effect <- FALSE
+    recruitment$logit_steep[1]$estimated <- FALSE
+    # turn on estimation of deviations
+    # recruit deviations should enter the model in normal space.
+    # The log is taken in the likelihood calculations
+    # alternative setting: recruitment$log_devs <- rep(0, length(om_input$logR.resid))
+    recruitment$log_devs <- methods::new(ParameterVector, om_input$logR.resid[-1], om_input$nyr-1)
+
+    # set up logR_sd using the normal log_sd parameter
+    recruitment_distribution <- 
+        new_process_distribution(par = "log_devs", module = recruitment, family = gaussian(),
+                                sd = list(value = om_input$logR_sd, estimated = FALSE),
+                                is_random_effect = FALSE)
+    recruitment$estimate_log_devs <- TRUE
+
+    expect_equal(log(om_input$logR_sd), recruitment_distribution$log_sd[1]$value)
+    expect_equal(length(recruitment$log_devs), length(recruitment_distribution$x))
+    expect_equal(length(recruitment_distribution$x), length(recruitment_distribution$expected_values))
+    expect_error(new_process_distribution(par = "log_devs", module = recruitment, family = multinomial(),
+                                sd = list(value = om_input$logR_sd, estimated = FALSE),
+                                is_random_effect = FALSE))
+    clear()
+})
+
+
+test_that("test new_data_distribution", {
+  # Data
+  catch <- em_input$L.obs$fleet1
+  # set fishing fleet catch data, need to set dimensions of data index
+  # currently FIMS only has a fleet module that takes index for both survey index and fishery catch
+  fishing_fleet_index <- new(Index, om_input$nyr)
+  fishing_fleet_index$index_data <- catch
+  fishing_fleet <- new(Fleet)
+  fishing_fleet$nages <- om_input$nages
+  fishing_fleet$nyears <- om_input$nyr
+  fishing_fleet$log_Fmort <- methods::new(ParameterVector, log(om_output$f), om_input$nyr)
+  fishing_fleet$log_Fmort$set_all_estimable(TRUE)
+  fishing_fleet$log_q[1]$value <- log(1.0)
+  fishing_fleet$estimate_q <- FALSE
+  fishing_fleet$random_q <- FALSE
+  fishing_fleet$SetObservedIndexData(fishing_fleet_index$get_id())
+
+  # Set up fishery index data using the lognormal
+  fishing_fleet_index_distribution <-
+    new_data_distribution(data_type = "cpue", module = fishing_fleet,
+                        family = lognormal(link = "log"),
+                        sd = list(value = rep(sqrt(log(em_input$cv.L$fleet1^2 + 1)), om_input$nyr),
```

Comment:
This code has been repeated several places. Maybe extract it into a separate variable.

Reply:
done